source/common/common/key_value_store_base.cc

```diff
@@ -29,9 +29,14 @@ absl::optional<absl::string_view> getToken(absl::string_view& contents, std::str
 } // namespace
 
 KeyValueStoreBase::KeyValueStoreBase(Event::Dispatcher& dispatcher,
-                                     std::chrono::seconds flush_interval)
-    : flush_timer_(dispatcher.createTimer([this]() { flush(); })) {
-  flush_timer_->enableTimer(flush_interval);
+                                     std::chrono::milliseconds flush_interval)
+    : flush_timer_(dispatcher.createTimer([this, flush_interval]() {
+        flush();
+        flush_timer_->enableTimer(flush_interval);
+      })) {
+  if (flush_interval.count() > 0) {
+    flush_timer_->enableTimer(flush_interval);
+  }
 }
```

Comment:
For my understanding: how was the timer enabled after flushing before this change?

Reply:
it wasn't - the bugfix is called out in PR description :-(